crml/nft/src/lib.rs

```diff
@@ -0,0 +1,651 @@
+		DirectSaleComplete(CollectionId, TokenId, AccountId, AssetId, Balance),
+		/// A direct sale has closed without selling
+		DirectSaleClosed(CollectionId, TokenId),
+		/// An auction has opened (collection, token, payment asset, reserve price)
+		AuctionOpen(CollectionId, TokenId, AssetId, Balance),
+		/// An auction has sold (collection, token, payment asset, bid, new owner)
+		AuctionSold(CollectionId, TokenId, AssetId, Balance, AccountId),
+		/// An auction has closed without selling (collection, token, reason)
+		AuctionClosed(CollectionId, TokenId, Reason),
+		/// A new highest bid was placed (collection, token, amount)
+		Bid(CollectionId, TokenId, Balance),
+	}
+);
+
+decl_error! {
+	/// Error for the staking module.
+	pub enum Error for Module<T: Trait> {
+		/// A collection with the same ID already exists
+		CollectionIdExists,
+		/// Given collection ID is not valid utf-8
+		CollectionIdInvalid,
+		/// No more Ids are available, they've been exhausted
+		NoAvailableIds,
+		/// Max tokens issued
+		MaxTokensIssued,
+		/// Too many attributes in the provided schema or data
+		SchemaMaxAttributes,
+		/// Provided attributes do not match the collection schema
+		SchemaMismatch,
+		/// Provided attribute is not in the collection schema
+		UnknownAttribute,
+		/// The provided attributes or schema cannot be empty
+		SchemaEmpty,
+		/// The schema contains an invalid type
+		SchemaInvalid,
+		/// The schema contains a duplicate attribute name
+		SchenmaDuplicateAttribute,
+		/// Given attirbute value is larger than the configured max.
+		MaxAttributeLength,
+		/// origin does not have permission for the operation
+		NoPermission,
+		/// The NFT collection does not exist
+		NoCollection,
+		/// The NFT does not exist
+		NoToken,
+		/// The NFT is not listed for a direct sale
+		NotForDirectSale,
+		/// The NFT is not listed for auction sale
+		NotForAuction,
+		/// Cannot operate on a listed NFT
+		TokenListingProtection,
+		/// Internal error during payment
+		InternalPayment,
+		/// Total royalties would exceed 100% of sale
+		RoyaltiesOvercommitment,
+		/// Auction bid was lower than reserve or current highest bid
+		BidTooLow
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Nft {
+		/// Map from collection to owner address
+		pub CollectionOwner get(fn collection_owner): map hasher(blake2_128_concat) CollectionId => Option<T::AccountId>;
+		/// Map from collection to its schema definition
+		pub CollectionSchema get(fn collection_schema): map hasher(blake2_128_concat) CollectionId => Option<NFTSchema>;
+		/// Map from collection to it's defacto royalty scheme
+		pub CollectionRoyalties get(fn collection_royalties): map hasher(blake2_128_concat) CollectionId => Option<RoyaltiesSchedule<T::AccountId>>;
+		/// Map from a token to it's royalty scheme
+		pub TokenRoyalties get(fn token_royalties): double_map hasher(blake2_128_concat) CollectionId, hasher(twox_64_concat) T::TokenId => Option<RoyaltiesSchedule<T::AccountId>>;
```

Comment:
Same here, the default RoyaltiesSchedule is distinct enough. So I'd be on the side of no need for the return value to be an option.